test/SILOptimizer/move_function_kills_copyable_values.sil

```diff
@@ -0,0 +1,90 @@
+  %13 = struct $Int (%11 : $Builtin.Int64)
+  %14 = integer_literal $Builtin.IntLiteral, 1024
+  %15 = builtin "s_to_s_checked_trunc_IntLiteral_Int64"(%14 : $Builtin.IntLiteral) : $(Builtin.Int64, Builtin.Int1)
+  (%16, %17) = destructure_tuple %15 : $(Builtin.Int64, Builtin.Int1)
+  %18 = struct $Int (%16 : $Builtin.Int64)
+  %19 = alloc_stack $(lower: Int, upper: Int)
+  %20 = tuple_element_addr %19 : $*(lower: Int, upper: Int), 0
+  %21 = tuple_element_addr %19 : $*(lower: Int, upper: Int), 1
+  store %13 to [trivial] %20 : $*Int
+  store %18 to [trivial] %21 : $*Int
+  %24 = alloc_stack $(lower: Int, upper: Int)
+  %25 = tuple_element_addr %24 : $*(lower: Int, upper: Int), 0
+  %26 = tuple_element_addr %24 : $*(lower: Int, upper: Int), 1
+  store %13 to [trivial] %25 : $*Int
+  store %18 to [trivial] %26 : $*Int
+  %29 = struct_element_addr %8 : $*Range<Int>, #Range.lowerBound
+  store %13 to [trivial] %29 : $*Int
+  destroy_addr %26 : $*Int
+  dealloc_stack %24 : $*(lower: Int, upper: Int)
+  %33 = alloc_stack $(lower: Int, upper: Int)
+  %34 = tuple_element_addr %33 : $*(lower: Int, upper: Int), 0
+  %35 = tuple_element_addr %33 : $*(lower: Int, upper: Int), 1
+  store %13 to [trivial] %34 : $*Int
+  store %18 to [trivial] %35 : $*Int
+  %38 = struct_element_addr %8 : $*Range<Int>, #Range.upperBound
+  store %18 to [trivial] %38 : $*Int
+  destroy_addr %34 : $*Int
+  dealloc_stack %33 : $*(lower: Int, upper: Int)
+  dealloc_stack %19 : $*(lower: Int, upper: Int)
+  %43 = struct $Range<Int> (%13 : $Int, %18 : $Int)
+  %44 = alloc_stack $Range<Int>
+  store %43 to [trivial] %44 : $*Range<Int>
+  %46 = function_ref @$sSlss16IndexingIteratorVyxG0B0RtzrlE04makeB0ACyF : $@convention(method) <τ_0_0 where τ_0_0 : Collection, τ_0_0.Iterator == IndexingIterator<τ_0_0>> (@in τ_0_0) -> @out IndexingIterator<τ_0_0>
+  %47 = apply %46<Range<Int>>(%7, %44) : $@convention(method) <τ_0_0 where τ_0_0 : Collection, τ_0_0.Iterator == IndexingIterator<τ_0_0>> (@in τ_0_0) -> @out IndexingIterator<τ_0_0>
+  dealloc_stack %44 : $*Range<Int>
+  dealloc_stack %8 : $*Range<Int>
+  br bb1
+
+bb1:
+  %51 = alloc_stack $Optional<Int>
+  %52 = begin_access [modify] [static] %7 : $*IndexingIterator<Range<Int>>
+  %53 = function_ref @$ss16IndexingIteratorV4next7ElementQzSgyF : $@convention(method) <τ_0_0 where τ_0_0 : Collection> (@inout IndexingIterator<τ_0_0>) -> @out Optional<τ_0_0.Element>
+  %54 = apply %53<Range<Int>>(%51, %52) : $@convention(method) <τ_0_0 where τ_0_0 : Collection> (@inout IndexingIterator<τ_0_0>) -> @out Optional<τ_0_0.Element>
+  end_access %52 : $*IndexingIterator<Range<Int>>
+  dealloc_stack %51 : $*Optional<Int>
+  switch_enum undef : $Optional<Int>, case #Optional.some!enumelt: bb2, case #Optional.none!enumelt: bb3
+
+bb2(%58 : $Int):
+  %59 = copy_value %2 : $Klass
+  %60 = begin_borrow [lexical] %59 : $Klass // expected-note {{use here}}
+  debug_value %60 : $Klass, let, name "m"
+  end_borrow %60 : $Klass
+  destroy_value %59 : $Klass
```

Comment:
This destroy_value was the one that we were emitting a note about incorrectly. In source it looked like a use on the '}' brace of a for loop.